### DISCUSSION TITLE
Bug raising

### DISCUSSION THREAD

**Discussion by ghost** [MarcEscandell/ALPypeRL#4] [General]
Fantastic job! I feel it will offer great help to me.

![pic](https://github.com/MarcEscandell/ALPypeRL/assets/118615882/9fb72b27-4425-44c8-aae7-de486333af8c)
However, I've got a bug here when I run the CartPole example. It seems there's some NullPointerException problem with the main.aLPypeRLConnector.requestAction() when I try to connect to the ALPypeRLConnector. Can you offer some insights please?

By the way, I'm not sure if there is any problem potentially caused by the mismatch of versions of the tools. Would you please kindly offer the verified version matches of Python packages and AnyLogic by which you have run correctly?

**Comment by MarcEscandell** [comment]
Hi @songzx-ll!
First of all, thanks for your kind words. I apologize for my late reply. 
You just got me in the middle of a new release for this project addresses some bugs and compatibility issues. I plan to release it in 1-2 days (if you don't mind, I'd recommend you to wait and try out again then). 
A bug that has been reported by multiple people is due to version compatibility. However, the exception thrown is different than yours (I plan to add some screenshots in the documentation).
Meanwhile, would you mind to detail out the steps that take you to this error so I can try to reproduce it on my side? What AnyLogic and python versions are you using?

**Reply by ghost** [reply]
I believe that I just took the steps as instructed, and upon running the simlation, the error pops up.
AnyLogic Version: 8.7.12 Professional; Python: python 3.11.4, tensorflow 2.15.0, gymnasium 0.28.1, ray 2.8.0, pydantic 2.5.1, py4j 0.10.9.7.

**Reply by ghost** [reply]
keras 2.15.0

**Reply by MarcEscandell** [reply]
I have pushed the new version now which I was testing with python 3.10. The previous version (the one you used) was based on python 3.10 and ray 2.7. Now I was testing for ray 2.8.
What operating system are you using?

**Reply by ghost** [reply]
(My OS is Windows 64) The error was caused by my mistakes in modifying the AnyLogic model... Now thay the error has been eliminated, I have run into another error. When I run the CartPole simulation in AntLogic IDE, it terminates after 5 iterations. Do you have any ideas? Lots of thanks!
![image](https://github.com/MarcEscandell/ALPypeRL/assets/118615882/ccae7e83-28bf-46e6-ba28-2523441f276d)
[console log.pdf](https://github.com/MarcEscandell/ALPypeRL/files/13499719/console.log.pdf)

**Reply by ghost** [reply]
I'm utilizing ALPypeRL 0.0.18.

**Reply by MarcEscandell** [reply]
I think I recognize this error. If not mistaken, this is because I have updated the AnyLogic _ALPypeLibrary_ (the library that you added to the AnyLogic _Palette_). Here's the link: https://github.com/MarcEscandell/ALPypeRL/tree/main/bin. Take the new library from the right folder depending on your AnyLogic version (anylogic-8.8 or anylogic-8.7 folders).

Once you update it, you should see the following:
![image](https://github.com/MarcEscandell/ALPypeRL/assets/25199391/518d0733-43e1-42fb-b915-c2af436199c3)

**Reply by ghost** [reply]
Perfect! It runs correctly now. Thank you very much!
